src/server/saved_objects/service/create_saved_objects_service.ts

```diff
@@ -59,37 +76,33 @@ export function createSavedObjectsService(server) {
   };
 
   const repositoryProvider = new SavedObjectsRepositoryProvider({
-    index: server.config().get('kibana.index'),
-    mappings: server.getKibanaIndexMappingsDsl(),
+    index: config.get('kibana.index'),
+    mappings,
     onBeforeWrite,
   });
 
   const scopedClientProvider = new ScopedSavedObjectsClientProvider({
-    index: server.config().get('kibana.index'),
-    mappings: server.getKibanaIndexMappingsDsl(),
-    onBeforeWrite,
-    defaultClientFactory({
-      request,
-    }) {
+    defaultClientFactory({ request }) {
       const { callWithRequest } = server.plugins.elasticsearch.getCluster('admin');
-      const callCluster = (...args) => callWithRequest(request, ...args);
+
+      const callCluster = <T>(method: string, params?: any): Promise<T> =>
+        callWithRequest(request, method, params);
 
       const repository = repositoryProvider.getRepository(callCluster);
 
       return new SavedObjectsClient(repository);
-    }
+    },
   });
 
   return {
     SavedObjectsClient,
     SavedObjectsRepository,
-    getSavedObjectsRepository: (...args) =>
-      repositoryProvider.getRepository(...args),
-    getScopedSavedObjectsClient: (...args) =>
-      scopedClientProvider.getClient(...args),
-    setScopedSavedObjectsClientFactory: (...args) =>
-      scopedClientProvider.setClientFactory(...args),
-    addScopedSavedObjectsClientWrapperFactory: (...args) =>
-      scopedClientProvider.addClientWrapperFactory(...args),
+    getSavedObjectsRepository: bind(repositoryProvider, 'getRepository'),
```

Comment:
We can definitely rewrite methods like this in a way that doesn't require argument inference, but it requires exporting a lot more types from the different modules and I'm super excited about argument inference.